dns/ddclient/src/opnsense/scripts/ddclient/lib/account/digitalocean.py

```diff
@@ -0,0 +1,175 @@
+    INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY
+    AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+    AUTHOR BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY,
+    OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+    SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+    INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+    CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+    ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+    POSSIBILITY OF SUCH DAMAGE.
+"""
+import json
+import syslog
+import requests
+from . import BaseAccount
+
+
+class DigitalOcean(BaseAccount):
+    _priority = 65535
+
+    _services = {"digitalocean": "api.digitalocean.com"}
+
+    def __init__(self, account: dict):
+        super().__init__(account)
+
+    @staticmethod
+    def known_services():
+        return {"digitalocean": "DigitalOcean"}
+
+    @staticmethod
+    def match(account):
+        return account.get("service") in DigitalOcean._services
+
+    def execute(self):
+        if super().execute():
+            recordType = "AAAA" if str(self.current_address).find(":") > 1 else "A"
+
+            headers = {
+                "User-Agent": "OPNsense-dyndns",
+                "Authorization": "Bearer " + self.settings.get("password"),
+            }
+
+            url = "https://%s/v2/domains/%s/records" % (
+                self._services[self.settings.get("service")],
+                self.settings.get("zone"),
+            )
+
+            if self.is_verbose:
+                syslog.syslog(
+                    syslog.LOG_NOTICE,
+                    "Account %s current IP is %s (%s)"
+                    % (self.description, self.current_address, recordType),
+                )
+
+                syslog.syslog(
+                    syslog.LOG_NOTICE,
+                    "Account %s updating hostnames %s"
+                    % (self.description, self.settings.get("hostnames", "")),
+                )
+
+            # Update each hostname
+            for hostname in self.settings.get("hostnames", "").split(","):
+
+                if self.is_verbose:
+                    syslog.syslog(
+                        syslog.LOG_NOTICE,
+                        "Account %s getting record ID for hostname %s (%s)"
+                        % (self.description, hostname, recordType),
+                    )
+
+                request = {
+                    "url": url,
+                    "params": {"name": hostname, "type": recordType},
+                    "headers": headers,
+                }
+
+                # Get record ID
+                response = requests.get(**request)
+
+                try:
+                    payload = response.json()
+                except requests.exceptions.JSONDecodeError:
+                    payload = {}
```

Comment:
```suggestion
                    continue
```

probably better to bail here, otherwise it will crash later, doubting a bit how fatal this is (`return False` is the other option)